pylint/config/config_initialization.py

```diff
@@ -118,6 +120,13 @@ def _config_initialization(
     # Link the base Namespace object on the current directory
     linter._directory_namespaces[Path(".").resolve()] = (linter.config, {})
 
-    # parsed_args_list should now only be a list of files/directories to lint.
+    # parsed_args_list should now only be a list of inputs to lint.
     # All other options have been removed from the list.
-    return parsed_args_list
+    return list(
+        chain.from_iterable(
+            # NOTE: 'or [arg]' is needed to maintain backwards compatibility
+            # TODO: remove 'or [arg]' in 3.0
+            glob(arg, recursive=True) or [arg]
```

Comment:
Shouldn't we add a deprecation warning if the ``[arg]`` is necessary here ? To be fair I don't understand what's happening that well and how this is enabling globbing pattern.

Reply:
Sorry, I didn't get it regarding the deprecation warning? Technically the `or [arg]` is needed in only one case: a `file.py` was specified as the input but there's no `file.py` in existence for glob to find - this case is seen here and there in tests.

Reply:
We should expand the comment in this case then. The described case is an error in this case, and probably only catched later in the code currently.

Reply:
> The described case is an error in this case, and probably only catched later in the code currently.

If you mean that there's a mistake in test cases - could be, yet there are quite more than one case that rely on behavior of `pylint` when specifying non-existent files affect the log.

Any suggestions how to better phrase the comment?

Reply:
Sorry, I missed this comment somehow.

What I mean is: the case you are catching here is if the user passes an unknown file to `pylint`, potentially by simply misspelling it. 
I just checked again and with pylint's current implementation, a fatal message "No module named <...>.py" is issued. We should preserve this behavior, so we can't even change this `or [arg]` after updating to 3.0. 
If we removed it, the misspelled file would get silently ignored, right?

My proposal for the comment:
```python
# NOTE: 'or [arg]' is needed in the case the input file or directory does not exist and 'glob(arg)' cannot find anything.
# Without this we would not be able to output the fatal import error for this module later on, as it would get silently ignored.
```

Reply:
Reasonable, done!